avocado/core/future/settings.py

```diff
@@ -0,0 +1,364 @@
+
+    def _append_system_config(self):
+        self._all_config_paths.append(self._config_path_system)
+        configs = glob.glob(os.path.join(self._config_dir_system_extra,
+                                         '*.conf'))
+        for extra_file in configs:
+            self._all_config_paths.append(extra_file)
+
+    def _append_user_config(self):
+        if not os.path.exists(self._config_path_local):
+            self._create_empty_config()
+        self._all_config_paths.append(self._config_path_local)
+
+    def _create_empty_config(self):
+        try:
+            path.init_dir(self._config_dir_local)
+            with open(self._config_path_local, 'w') as config_local_fileobj:
+                content = ("# You can use this file to override "
+                           "configuration values from '%s and %s\n"
+                           % (self._config_path_system,
+                              self._config_dir_system_extra))
+                config_local_fileobj.write(content)
+        except IOError:     # Some users can't write it (docker)
+            pass
+
+    def _get_option_from_tag(self, tag):
+        try:
+            mapping = self._long_mapping[tag]
+            return self._namespaces[mapping]
+        except KeyError:
+            msg = "{} not found in internal mapping.".format(tag)
+            raise SettingsError(msg)
+
+    def _prepare_base_dirs(self):
+        def get_base_dirs():
+            if 'VIRTUAL_ENV' in os.environ:
+                cfg_dir = os.path.join(os.environ['VIRTUAL_ENV'], 'etc')
+                user_dir = os.environ['VIRTUAL_ENV']
+            else:
+                cfg_dir = '/etc'
+                user_dir = os.path.expanduser("~")
+            return cfg_dir, user_dir
+
+        cfg_dir, user_dir = get_base_dirs()
+
+        self._config_dir_system = os.path.join(cfg_dir, 'avocado')
+        self._config_dir_system_extra = os.path.join(cfg_dir,
+                                                     'avocado',
+                                                     'conf.d')
+        self._config_dir_local = os.path.join(user_dir, '.config', 'avocado')
+        self._config_path_system = os.path.join(self._config_dir_system,
+                                                'avocado.conf')
+        self._config_path_local = os.path.join(self._config_dir_local,
+                                               'avocado.conf')
+
+    def as_dict(self):
+        """Return an ordered dictionary with the current active settings.
+
+        This will return a ordered dict of both: configparse and merged
+        argparse options.
+        """
+        result = {}
+        for section in self._config.sections():
+            result[section] = dict(self._config.items(section))
+        return result
+
+    def get(self, section, key, key_type=str):
+        """Returns the current value inside a setion + key.
+
+        If this section do not exists on config files but it was registered
```

Comment:
This seems to be limited to the config files (it only uses the `ConfigParser` instance), so how will it get values registered with `register_option`?

Reply:
Not sure if I understood your question, but line 239 calls update_settings that will update `self._config` with argparse too.

Reply:
Yes, seems that I missed that.